csharp/src/Ice/Incoming.cs

```diff
@@ -461,7 +471,10 @@ public Ice.OutputStream writeEmptyParams()
         {
             if(_response)
             {
-                var os = new Ice.OutputStream(_instance, Ice.Util.currentProtocolEncoding);
+                // If there's an output stream set, re-use it for the response
+                var os = _os == null ? new Ice.OutputStream(_instance, Ice.Util.currentProtocolEncoding) : _os;
+                Debug.Assert(_os.pos() == 0);
+                _os = null;
```

Comment:
Why do we need to `_os = null;` here, shouldn't we just reset it?

Reply:
It's assumed to be reset already (that's what the asset checks). We set it to null here to indicate that it's now owned by the caller. The caller will in general pass it back to the Incoming object when it's calling one of the setResult methods.

Reply:
Nevermind, I see we are returning it so we cannot reset it...